src/module.c

```diff
@@ -4803,6 +4807,26 @@ int moduleAllDatatypesHandleErrors() {
     return 1;
 }
 
+/* Returns 0 if there's at least one registered data type that did not declare
+ * REDISMODULE_OPTIONS_HANDLE_REPL_ASYNC, in which case diskless async loading should be avoided
+ * because module doesn't know there can be traffic during database full resynchronization. */
+int moduleAllDatatypesHandleReplAsync() {
+    dictIterator *di = dictGetIterator(modules);
+    dictEntry *de;
+
+    while ((de = dictNext(di)) != NULL) {
+        struct RedisModule *module = dictGetVal(de);
+        if (listLength(module->types) &&
+            !(module->options & REDISMODULE_OPTIONS_HANDLE_REPL_ASYNC))
```

Comment:
@MeirShpilraien it occurred to me for a moment that maybe the "data type" part should be dropped, since modules with aux data can also be affected.
then i realized that we are lucky that these aux fields are only possible when data type being registered.

so i just wanna check with you that you don't see any problem with other modules, i.e. ones that use RM_Call or RM_OpenKey

Reply:
@oranagra I actually believe that we should not check if a module created a datatype here, a module might need to be aware of this feature even if it does not have a datatype. For example, a module that collects stats about the keyspace. Such module will probably register on keyspace events, it will collect stats and it will want to be aware of possible swap db to know how to handle loaded events correctly. I agree that without loaded events, only modules with datatypes need to be aware of this feature, but the loaded event breaks this assumption.

Reply:
ok. thanks.
so let's remove the `listLength(module->types)` part

Reply:
So I guess method name becomes `moduleHandleReplAsync`? (see change)

Reply:
maybe module**s**HandleReplAsync?

Reply:
actually, `module` is the prefix for all methods.
so maybe `moduleAllModulesHandleReplAsync`?